src/trackHttp.test.js

```diff
@@ -0,0 +1,85 @@
+import {trackPromise, emitter} from './trackHttp';
+import {Emitter} from './tinyemmiter';
+
+
+describe('trackPromise', () => {
+  test('On Initial case, promise fired, promise emitter.emit is called', () => {
+    // Arrange
+    emitter.emit = jest.fn((a,b) => {
+      return;
+    });
+
+    const myPromise =  Promise.resolve().then(() => {
+      return "ok";
+    });
+
+    // Act
+    trackPromise(myPromise);
+
+    // Assert
+    expect(emitter.emit).toHaveBeenCalledTimes(1);
+
+    return myPromise;
+  });
+
+  test('Promise tracked, we got resolve, check that emit is called 2 times', () => {
+    // Arrange
+    emitter.emit = jest.fn((a,b) => {
+      return;
+    });
+
+    const myPromise =  Promise.resolve();
+
+    // Act
+    trackPromise(myPromise);
+
+    // Assert
+    return myPromise.then(() => {
+      expect(emitter.emit).toHaveBeenCalledTimes(2);
+    });
+  });
+
+  test.skip('Promise tracked, we got fail, check that emit is called 2 times', () => {
```

Comment:
I would rename it to: "when tracking a single promise and after it is rejected, emit should have been called twice"